deepspeed/runtime/zero/partition_parameters.py

```diff
@@ -1501,23 +1501,22 @@ def _partition_param_sec(self, param, buffer=None, has_been_updated=False):
             partition_size = tensor_size // self.dp_world_size
 
             secondary_partition_size = int(tensor_size // self.num_ranks_in_param_group)
-            if param.ds_secondary_tensor is None:
-                final_location = None
-                secondary_partitioned_tensor = torch.empty(secondary_partition_size,
-                                                           dtype=param.dtype,
-                                                           device=self.remote_device)
-
-                if self.pin_memory:
-                    secondary_partitioned_tensor = secondary_partitioned_tensor.pin_memory()
-                # quantize the tensor if it's not trainable
-                if not param.requires_grad and self.quantized_nontrainable_weights:
-                    secondary_partitioned_tensor, secondary_partitioned_tensor.ds_quant_scale = self.quantizer_module.quantize(
-                        secondary_partitioned_tensor)
-                secondary_partitioned_tensor.requires_grad = False
-                param.ds_secondary_tensor = secondary_partitioned_tensor
-                param.ds_secondary_tensor.ds_numel = secondary_partition_size
-                param.ds_secondary_tensor.status = PartitionedParamStatus.AVAILABLE
-                param.ds_secondary_tensor.final_location = final_location
+            final_location = None
+            secondary_partitioned_tensor = torch.empty(secondary_partition_size,
```

Comment:
@HeyangQin, perhaps I misunderstood our chat. I thought secondary tensor should only be created on a forward pass of a training iteration. Currently, partition_param is called on every forward and backward, which means secondary tensor is created in more cases than required. I would like to chat more about this. Thanks!